chapters/publication.tex

```diff
@@ -99,91 +174,259 @@ \section{Collaborating on academic writing}
+The purpose of releasing code is to allow others to understand
+exactly what you have done in order to obtain your results,
+as well as to apply similar methods in future projects.
+Therefore it should both be functional and readable.
+Code is often not written this way when it is first prepared,
+so it is important for you to review the content and organization
+so that a new reader can figure out what and how your code should do.
+Therefore, whereas your data should already be very clean at this stage,
+your code is much less likely to be so, and this is where you need to make
+time investments prior to releasing your replication package.
+By contrast, replication code usually has few legal and privacy constraints.
+In most cases code will not contain identifying information;
+check carefully that it does not.
+Publishing code also requires assigning a license to it;
+in a majority of cases, code publishers like GitHub
+offer extremely permissive licensing options by default.
+(If you do not provide a license, nobody can use your code!)
+
+Make sure the code functions identically on a fresh install of your chosen software.
+A new user should have no problem getting the code to execute perfectly.
+In either a scripts folder or in the root directory,
+include a master script (dofile or R script for example).
+The master script should allow the reviewer
+to change a single line of code: the one setting the directory path.
+After that, running the master script should run the entire project
+and re-create all the raw outputs exactly as supplied.
+Indicate the filename and line to change.
+Check that all your code will run completely on a new computer:
+Install any required user-written commands in the master script
+(for example, in Stata using \texttt{ssc install} or \texttt{net install}
+and in R include code giving users the option to install packages,
+including selecting a specific version of the package if necessary).
+In many cases you can even directly provide the underlying code
+for any user-installed packages that are needed to ensure forward-compatibility.
+Make sure system settings like \texttt{version}, \texttt{matsize}, and \texttt{varabbrev} are set.
+
+Finally, make sure that the code and its inputs and outputs are clearly identified.
+A new user should, for example, be able to easily identify and remove
+any files created by the code so that they can be recreated quickly.
+They should also be able to quickly map all the outputs of the code
+to the locations where they are placed in the associated published material,
+such as ensuring that the raw components of figures or tables are clearly identified.
+For example, outputs should clearly correspond by name to an exhibit in the paper, and vice versa.
+(Supplying a compiling \LaTeX\ document can support this.)
+Code and outputs which are not used should be removed -- 
+if you are using GitHub, consider making them available in a different branch for transparency.
```

Comment:
"_consider making them available in a different branch_" - we cannot just mention this without explaining. And keeping track of a stale branch without in a meaningful way i non-trivial, so I do not know if we should keep this here.

Reply:
fb7aca54df3015bb18dc1257c989c07df9e15792